server_info_test.go

```diff
@@ -0,0 +1,124 @@
+package git
+
+import (
+	"io"
+	"strings"
+
+	"github.com/go-git/go-billy/v5/memfs"
+	fixtures "github.com/go-git/go-git-fixtures/v4"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/storage/memory"
+	. "gopkg.in/check.v1"
+)
+
+func (s *RepositorySuite) TestUpdateServerInfoInit(c *C) {
+	fs := memfs.New()
+	st := memory.NewStorage()
+	r, err := Init(st, fs)
+	c.Assert(err, IsNil)
+	c.Assert(r, NotNil)
+
+	err = UpdateServerInfo(st, fs)
+	c.Assert(err, IsNil)
+}
+
+func testUpdateServerInfo(c *C, f *fixtures.Fixture) {
```

Comment:
Would be nice to add another test that asserts subsequent calls to update the server info work properly after new changes to the repository take place.